hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -106,6 +107,38 @@
       GC_ENABLED, "external.table.purge"
   );
 
+  private static Map<Long, String> tableLocksById;
+  private static ClientPool<IMetaStoreClient, TException> globalMetaClients;
+
+  /**
+   * Initialize a hook to avoid that the finally code block used to unlock hive table may not be executed
+   * due to the following conditions:
+   *   - System.exit(N)
+   *   - all non-daemon threads exit
+   *
+   * Also, to avoid memory leaks caused by addShutdownHook, we use a class level hook.
+   */
+  private static synchronized void initUnlockTableHook(ClientPool metaClients) {
+    globalMetaClients = metaClients;
```

Comment:
What happens if you have multiple catalogs configured with different URI?
Having a global variable storing the clients could be a problem

Reply:
+1. This is a use case that I know exists at some companies (multiple catalogs configured with different URIs).

Reply:
Thanks. I've fixed it.